daemon/routing.h

```diff
@@ -20,6 +20,11 @@ struct node_connection {
 
 struct node {
 	struct pubkey id;
+
+	/* IP/Hostname and port of this node */
+	char *hostname;
```

Comment:
const char *hostname.  C's "free()" doesn't accept const pointers, but tal_free() does.  It might feel a bit weird at first, but lifetime and changeability are conceptually different.